src/models/tools/tool-content-info.ts

```diff
@@ -54,6 +57,10 @@ const gToolContentInfoMapByTool: IToolContentInfoMap = {};
 export function registerToolContentInfo(toolContentInfo: IToolContentInfo) {
   gToolContentInfoMapById[toolContentInfo.id] = toolContentInfo;
   gToolContentInfoMapByTool[toolContentInfo.tool] = toolContentInfo;
+  const { icon } = toolContentInfo;
+  if (icon) {
+    appIcons[`icon-${toolContentInfo.tool}-tool`] = icon;
+  }
```

Comment:
This should be double-checked with the chat panel's tile-specific icon functionality. It should still work because of the indirection through `appIcons`, but if it's going to break anywhere that'd be a likely place. In the longer term we should be able to get rid of tool icons in `appIcons` altogether and just look them up from the registry when necessary, I would think.

Reply:
Yes it looks like it should work fine:
https://github.com/concord-consortium/collaborative-learning/blob/928587444a08aacc304b824f380ac8ac7f571611/src/components/chat/tool-icon-component.tsx#L14-L16

I agree it would be better to just get the icon directly out fo the registry. I wasn't sure the best way to do that because the icon list is both tool icons and general application icons. But I think I see a way to do it now. I'll give that a shot...

Reply:
>I agree it would be better to just get the icon directly out of the registry. I wasn't sure the best way to do that because the icon list is both tool icons and general application icons.

I was thinking that the general application icons could stay in the `appIcons` structure and the tool icons could be gotten through the registry. I suspect clients generally know which of the two they're trying to access at a given time.

Reply:
Yup that is what I did. See what you think about the new approach.